### PR TITLE
[e2e][CL]: pool migration 

### PR DESCRIPTION
**Pull request by ThanhNhann**
Closes: #4496 

## What is the purpose of the change

Add e2e test pool migrations from balancer/stableswap to CL

## Brief Changelog
- Add e2e test pool migrations from balancer/stableswap to CL with 3 steps:
    - Submit a gov proposal to update pools allowed to migrate
    - Migrate your position from balancer / stableswap to CL
    - Validate that the position migrated correctly

## Testing and Verifying

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added unit test that validates ...*
  - *Added integration tests for end-to-end deployment with ...*
  - *Extended integration test for ...*
  - *Manually verified the change by ...*

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Comment by czarcas7ic** [comment]
@ThanhNhann just wanted to check in to see what the plan for this was

**Comment by ThanhNhann** [comment]
Hi @czarcas7ic, thanks for your ping, I'm updating it

File: CHANGELOG.md
Comment (czarcas7ic) [inline CHANGELOG.md:58]: This isn't needed

File: tests/e2e/configurer/chain/commands.go
Comment (czarcas7ic) [inline tests/e2e/configurer/chain/commands.go:182]: I think this belongs in the queries.go file

File: tests/e2e/e2e_test.go
Comment (czarcas7ic) [inline tests/e2e/e2e_test.go:1708]: This probably shouldn't be a fail. It should just be a check to see if its enabled, and if its not then submits the param change prop

File: tests/e2e/helpers_e2e_test.go
Comment (mattverse) [inline tests/e2e/helpers_e2e_test.go:210]:
```suggestion
```

File: tests/e2e/helpers_e2e_test.go
Comment (mattverse) [inline tests/e2e/helpers_e2e_test.go:288]:
```suggestion
func (s *IntegrationTestSuite) testPoolMigration(
```

**Review by mattverse — APPROVED** [review]
Some nit comments, I think this lgtm! Nice work 🔥

**Comment by mattverse** [comment]
@ThanhNhann Can you leave more detailed in line comments for the test on what we are testing and what are the things we expect to be tested through this test?

**Comment by czarcas7ic** [comment]
@ThanhNhann Is there still intent to resolve issues for this PR?

**Comment by ThanhNhann** [comment]
yeah, I will update the code with @mattverse review today @czarcas7ic

File: tests/e2e/scripts/nativeDenomPoolMigration.json
Comment (ThanhNhann) [reply tests/e2e/scripts/nativeDenomPoolMigration.json]: thank matt, I just forgot to update it after done the test

File: tests/e2e/configurer/chain/commands.go
Comment (ThanhNhann) [reply tests/e2e/configurer/chain/commands.go:765]: yeah, good idea @mattverse, we also have another func like above [here](https://github.com/osmosis-labs/osmosis/blob/d459a93347d836d243692cfa923932d598c6cb30/tests/e2e/configurer/chain/commands.go#L491), so have a helper func like you said is great. I will create the pr for this idea after this pr was merged

**Comment by czarcas7ic** [comment]
@ThanhNhann please ping when r4r, thanks!

**Comment by ThanhNhann** [comment]
It's good to review now @czarcas7ic

File: tests/e2e/helpers_e2e_test.go
Comment (mattverse) [inline tests/e2e/helpers_e2e_test.go:349]: can we also check the amount in the new lock?

File: tests/e2e/helpers_e2e_test.go
Comment (mattverse) [inline tests/e2e/helpers_e2e_test.go:358]: Can we do more detailed checks for the unbonding case? 
- amount left in lock
- amount delegated

File: tests/e2e/e2e_test.go
Comment (mattverse) [inline tests/e2e/e2e_test.go:1751]: what are the differences between these two cases?

**Review by mattverse — COMMENTED** [review]
LGTM, I have left some last comments that would be best if addressed, after that I think we should be good for merge

**Comment by czarcas7ic** [comment]
@ThanhNhann would like the point about 6 cases vs 4 addressed, then I think we are good to merge.

**Comment by ThanhNhann** [comment]
thanks your review @czarcas7ic, I will update soon

File: tests/e2e/helpers_e2e_test.go
Comment (ThanhNhann) [reply tests/e2e/helpers_e2e_test.go:350]: Updated

File: tests/e2e/configurer/chain/commands.go
Comment (ThanhNhann) [reply tests/e2e/configurer/chain/commands.go:765]: Yeah its here https://github.com/osmosis-labs/osmosis/issues/5601

File: tests/e2e/e2e_test.go
Comment (ThanhNhann) [reply tests/e2e/e2e_test.go]: Added @czarcas7ic

File: tests/e2e/e2e_test.go
Comment (czarcas7ic) [inline tests/e2e/e2e_test.go:1700]: Update godoc please

Comment (ThanhNhann) [reply tests/e2e/e2e_test.go]: Updated

**Review by czarcas7ic — COMMENTED** [review]
Nice work on this! I want to do one more review before merging though

File: tests/e2e/helpers_e2e_test.go
Comment (ThanhNhann) [reply tests/e2e/helpers_e2e_test.go:201]: balanceAfterJoin will have another coins so I think we should use SafeSub here (I double checked this)

File: tests/e2e/helpers_e2e_test.go
Comment (ThanhNhann) [reply tests/e2e/helpers_e2e_test.go]: Updated

**Comment by czarcas7ic** [comment]
Hey, sorry for tall the back and forths on this. It looks like this test is significantly longer when compared to all the other E2E tests:

```
    --- PASS: TestIntegrationTestSuite/TestAConcentratedLiquidity_CanonicalPool_And_Parameters (0.46s)
    --- PASS: TestIntegrationTestSuite/TestAddToExistingLock (39.69s)
    --- PASS: TestIntegrationTestSuite/TestAddToExistingLockPostUpgrade (2.63s)
    --- SKIP: TestIntegrationTestSuite/TestArithmeticTWAP (0.00s)
    --- PASS: TestIntegrationTestSuite/TestConcentratedLiquidity (93.36s)
    --- PASS: TestIntegrationTestSuite/TestCreateConcentratedLiquidityPoolVoting_And_TWAP (42.77s)
    --- PASS: TestIntegrationTestSuite/TestExpeditedProposals (21.58s)
    --- PASS: TestIntegrationTestSuite/TestGeometricTWAP (12.51s)
    --- PASS: TestIntegrationTestSuite/TestGeometricTwapMigration (4.51s)
    --- PASS: TestIntegrationTestSuite/TestIBCTokenTransferAndCreatePool (60.43s)
    --- PASS: TestIntegrationTestSuite/TestIBCTokenTransferRateLimiting (61.23s)
    --- PASS: TestIntegrationTestSuite/TestIBCWasmHooks (8.40s)
    --- PASS: TestIntegrationTestSuite/TestLargeWasmUpload (4.03s)
    --- PASS: TestIntegrationTestSuite/TestPacketForwarding (10.96s)
    --- PASS: TestIntegrationTestSuite/TestPoolMigration (483.99s)
    --- PASS: TestIntegrationTestSuite/TestProtoRev (45.92s)
    --- PASS: TestIntegrationTestSuite/TestStableSwapPostUpgrade (6.62s)
    --- PASS: TestIntegrationTestSuite/TestStateSync (52.40s)
    --- PASS: TestIntegrationTestSuite/TestSuperfluidVoting (39.96s)
    ```
    
    We might need to add some randomized logic to select a single migration path rather than checking every one every time. That or add some go routines in the test itself. 100s I think should be the max for one of these tests, and this is almost 5x that (not at all your fault, its a hefty test, just need to make this shorter)

**Pull request closed without merging**
